zerver/decorator.py

```diff
@@ -183,11 +183,17 @@ def validate_api_key(request, role, api_key, is_webhook=False):
     # Remove whitespace to protect users from trivial errors.
     role, api_key = role.strip(), api_key.strip()
 
-    if is_remote_server(role):
+    if settings.ZILENCER_ENABLED and is_remote_server(role):
```

Comment:
Doesn't this change the logic for the case where ZILENCER_ENABLED is False but is_remote_server is True?